pkg/feature/servicemesh/cleanup.go

```diff
@@ -55,7 +56,11 @@ func RemoveExtensionProvider(controlPlane infrav1.ControlPlaneSpec, extensionNam
 		}
 
 		if removed {
-			return cli.Update(ctx, smcp)
+			// Update the ServiceMeshControlPlane with the removed extension provider.
+			// As it could have been updated by another controller in the meantime, we need to retry on conflict.
+			return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
It's an additional fix, right?

Reply:
Yes, discovered during testing of this change. We can extract it to a tiny PR but it's a bit overdoing in my eyes.

Reply:
I mostly wanted to confirm that I did not myself miss the point from the patch :)

Reply:
PS: and it deserve to be separated just because it is not part of the PR's change, not because of the size
/me hides :)))